index.js

```diff
@@ -20,7 +20,24 @@ function fastifyStatic (fastify, opts, next) {
   })
   if (error !== undefined) return next(error)
 
-  const root = opts.root
+  const setHeaders = opts.setHeaders
+
+  if (setHeaders !== undefined && typeof setHeaders !== 'function') {
+    return next(new TypeError('The `setHeaders` option must be a function'))
+  }
+
+  const sendOptions = {
+    root: opts.root,
+    acceptRanges: opts.acceptRanges,
+    cacheControl: opts.cacheControl,
+    dotfiles: opts.dotfiles,
+    etag: opts.etag,
+    extensions: opts.extensions,
+    immutable: opts.immutable,
+    index: opts.index,
+    lastModified: opts.lastModified,
+    maxAge: opts.maxAge || 0
+  }
```

Comment:
Can you add a unit test (maybe using proxyquire)? So we can test that the options are passed through.